samza-core/src/test/scala/org/apache/samza/container/TestSamzaContainer.scala

```diff
@@ -126,6 +126,45 @@ class TestSamzaContainer extends AssertionsForJUnit with MockitoSugar {
     verify(this.runLoop).run()
   }
 
+
+  @Test
+  def testShutDownSequenceForStandbyContainers() {
+    class ShutDownSignal(container: SamzaContainer) extends Runnable {
+      def run(): Unit = {
+        Thread.sleep(2000)
```

Comment:
Can we do without the sleep in the test ?  Also, we should verify that the countdown() was actually called.
I think it should be possible to test the behavior by making the latch visible for testing ?

Reply:
runLoop.run is blocking call so cannot do this without sleep

> Also, we should verify that the countdown() was actually called

well the verification is done on the shutdown methods which only gets called once that latch is counted down if the latch was not all these shutdown methods won't be invoked, hence I do not feel exposing this latch outside and verifying on it